app/page.tsx

```diff
@@ -1,18 +1,38 @@
 'use client';
 
+import React, { useState } from 'react';
+
 import { UserButton, useUser } from '@clerk/nextjs';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { httpBatchLink } from '@trpc/client';
 import { Group } from '@mantine/core';
+
 import { Welcome } from '../components/Welcome/Welcome';
 import { Chat } from './chat';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
-
-const queryClient = new QueryClient();
+import { trpc } from '@/utils/trpc';
 
 export default function HomePage() {
   const { isLoaded, user } = useUser();
+  const [queryClient] = useState(() => new QueryClient());
+  const [trpcClient] = useState(() =>
+    trpc.createClient({
+      links: [
+        httpBatchLink({
+          url: 'http://localhost:3000/trpc',
+          // You can pass any HTTP headers you wish here
+          async headers() {
+            return {
+              authorization: getAuthCookie(),
```

Comment:
This method is undefined and is a placeholder from the tRPC docs. This will probably be the next error you have to tangle with - I _think_ you can just drop it, but Clerk might want you to pull something in here too. 🕵️